### ISSUE TITLE
Exception not handled by on error event

### ISSUE THREAD

**Issue by sakura26** [lucagrulla/node-tail#54]
If file watched deleted, node will exit with exception and can't be catched outside

```
fs.js:954
  return binding.stat(pathModule._makeLong(path));
                 ^

Error: ENOENT: no such file or directory, stat '/tmp/1234.csv'
    at Object.fs.statSync (fs.js:954:18)
    at Tail.watchEvent (/opt/JAICAS/node_modules/tail/lib/tail.js:123:18)
    at FSWatcher.<anonymous> (/opt/JAICAS/node_modules/tail/lib/tail.js:105:24)
    at emitTwo (events.js:106:13)
    at FSWatcher.emit (events.js:192:7)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1367:12)
```

This is caused by 
https://github.com/lucagrulla/node-tail/blob/master/src/tail.coffee#L62

you can simply fix it with try/cache or check file exist before fs.statSync(this.filename);

**Comment by lucagrulla** [comment]
Thanks for finding this. 
I'd say that between the 2 options you suggested check file existence is neater. 
Are you planning to send a PR with the fix?

**Comment by lucagrulla** [comment]
fixed with release 1.2.3
